### PR TITLE
Game logic Completion

### PR DESCRIPTION
**Pull request by Aaron-RN**

### CODE REVIEW

File: lib/game_logic.rb [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+  def move_made(player, pos)
+    @board[pos - 1] = player.symbol.to_s.green if player.color == 'green'
+    @board[pos - 1] = player.symbol.to_s.red if player.color == 'red'
+    @board[pos - 1] = player.symbol.to_s.blue if player.color == 'blue'
+    @turns_taken += 1
+    game_over?
+  end
+
+  def draw?
+    puts 'DRAW!'.bold.underline,"\n" if @turns_taken >= @max_turns
```
Comment (bolah2009) [inline lib/game_logic.rb]: Let's keep user interaction away from game logic.

> `main.rb` file should be responsible for user interaction. This is the only one place where `puts` and `get.chomp` can be used.

File: lib/game_logic.rb [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    winning_combo.each do |item|
+      game_won(@player_one) if item.all? { |e| @player_one.positions_mark.include?(e) }
+      game_won(@player_two) if item.all? { |e| @player_two.positions_mark.include?(e) }
+    end
+
+    draw? unless @game_over
+  end
+end
+
+class Player
```
Comment (bolah2009) [inline lib/game_logic.rb]: Can we move this `Player class` to a new file instead? Maybe `player.rb`... :thinking:

**Review by bolah2009 — CHANGES_REQUESTED** [review]
## Nice user interface :clap: 

Please check out the comments below and make changes, also, let's try and keep the one class from doing another class function.

#### :recycle:  Make these changes and submit another code review request. :+1: 

#### Feel free to reach me on slack if you having issues implementing these features. :+1: 

Good luck with your next submission. :star2: 

Cheers
[Bola](https://bolabuari.com/)

###### PS: Note that comments tagged as `OPTIONAL` are not part of the minimum requirements to approve this PR

**Review by BalintMendli — APPROVED** [review]
Hi @Aaron-RN @zhuchini ,

Great job implementing requested changes. 👏 

### 👍 

Status: **Approved**

**Pull request merged**